rviz_default_plugins/src/rviz_default_plugins/displays/marker/markers/triangle_list_marker.cpp

```diff
@@ -249,6 +296,26 @@ void TriangleListMarker::updateMaterial(
     material_->getTechnique(0)->setSceneBlending(Ogre::SBT_REPLACE);
     material_->getTechnique(0)->setDepthWriteEnabled(true);
   }
+
+  if (hasTexture(new_message)) {
+    // If the texture is marked as updated, delete the last texture (if it exists).
+    Ogre::ResourcePtr texture = Ogre::TextureManager::getSingleton().getByName(
+      new_message->texture_map, "rviz_rendering");
+    if (new_message->texture_update) {
+      if (texture != NULL) {
+        Ogre::TextureManager::getSingleton().remove(texture);
+      }
+    }
+    if (Ogre::TextureManager::getSingleton().getByName(
+        new_message->texture_map,
+        "rviz_rendering") == NULL)
+    {
+      loadImage(new_message->texture_map, new_message->texture_map);
```

Comment:
Considering changing the signature to `loadImage(texture_uri)`, since we're using the URI as the name as well. Unless there's plans to actually name the texture otherwise.